src/imio/events/core/tests/test_indexer.py

```diff
@@ -0,0 +1,105 @@
+from imio.events.core.testing import IMIO_EVENTS_CORE_INTEGRATION_TESTING
+from plone import api
+from plone.app.testing import setRoles
+from plone.app.testing import TEST_USER_ID
+from zope.component import getUtility
+from zope.schema.interfaces import IVocabularyFactory
+
+import unittest
+
+
+class TestIndexer(unittest.TestCase):
+
+    layer = IMIO_EVENTS_CORE_INTEGRATION_TESTING
+
+    def setUp(self):
+        self.portal = self.layer["portal"]
+        self.portal_catalog = api.portal.get_tool("portal_catalog")
+
+        setRoles(self.portal, TEST_USER_ID, ["Manager"])
+        self.entity = api.content.create(
+            container=self.portal,
+            type="imio.events.Entity",
+            id="imio.events.Entity",
+            local_categories="Foo\r\nbaz\r\nbar",
+        )
+        self.agenda = api.content.create(
+            container=self.entity,
+            type="imio.events.Agenda",
+            id="imio.events.Agenda",
+        )
+
+    def _search_all_from_vocabulary(self, vocabulary, context, catalog):
+        factory = getUtility(
+            IVocabularyFactory,
+            vocabulary,
+        )
+        output = {}
+        vocabulary = factory(context)
+        for v in vocabulary.by_value:
+            result = catalog.searchResults(**{"category_and_topics": v})
+            if len(result) == 0:
+                continue
+            output[v] = [r.getObject().id for r in result]
+
+        return output
+
+    def test_event_with_nothing(self):
+        _ = api.content.create(
+            container=self.agenda,
+            type="imio.events.Event",
+            id="imio.events.Event",
+        )
+        search_result = self._search_all_from_vocabulary(
+            "imio.events.vocabulary.EventsCategoriesAndTopicsVocabulary",
+            self.agenda,
+            self.portal_catalog,
+        )
+
+        self.assertEqual(len(search_result), 0)
+
+    def test_event_with_one_of_each(self):
+        _ = api.content.create(
```

Comment:
Remove attribution of content creation if not necessary or use a comprehensible variable name